ratatool-examples/src/test/scala/com/spotify/ratatool/examples/ExampleAvroGenTest.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.ratatool.examples
+
+import java.util.UUID
+
+import com.spotify.ratatool.avro.specific.{EnumField, ExampleRecord}
+import org.scalacheck.{Gen, Prop, Properties}
+import org.scalacheck.Prop.{AnyOperators, BooleanOperators, forAll}
+
+import scala.collection.JavaConverters._
+
+object ExampleAvroGenTest extends Properties("ExampleAvroGenerator") {
+  val gen: Gen[ExampleRecord] = ExampleAvroGen.exampleRecordGen
+
+  property("round trips UUID") = forAll(gen) { m =>
+    UUID.fromString(m.getRecordId.toString).toString ?= m.getRecordId.toString
+  }
+
+  property("generates valid dependent int") = forAll(gen) { m =>
+    (m.getIndependentIntField == 0
+      && m.getDependentIntField == Int.MaxValue) :| "Max if indep is 0" ||
+    (m.getIndependentIntField != 0
+      && m.getDependentIntField == m.getIndependentIntField/2) :| "Half when indep is not 0"
+  }
+
+  property("generates valid dependent enum") = forAll(gen) { m =>
+    (m.getIndependentStringField.toString.startsWith("Exception") &&
+      m.getDependentEnumField == EnumField.Failure) :| "Is Failure on Exception" ||
+    (!m.getIndependentStringField.toString.startsWith("Exception") &&
+      m.getDependentEnumField == EnumField.Success) :| "Is Success when non-Exception"
+  }
+
+  property("double field within bounds") = forAll(gen) { m =>
+    m.getBoundedDoubleField <= 1.0 && m.getBoundedDoubleField >= -1.0
+  }
+
+  property("map field size within bounds") = forAll(gen) { m =>
```

Comment:
this is interesting

property based testing for a generator

should we recommend people to test their generators when they modify their own code using property based tests

Reply:
Potentially yes, I think it's dependent on the case and complexity.